src/types/index.ts

```diff
@@ -0,0 +1,213 @@
+type TransformedEvent = {
```

Comment:
As we are doing a great job of defining all the types, let's also add documentation for the relevant ones to improve maintainability.

Reply:
Yes we can but we have to think answers to some questions
- For whom will these docs be helpful
- Is it for external views as well
- Are we sure the type definitions we have defined are stable and not going to change any time soon?